articles/actions.md

```diff
@@ -0,0 +1,338 @@
+
+The purpose of this service is to get the final result of a service.
+After a goal has been accepted the client should call this service to receive the result.
+The result will indicate the final status of the goal and any user defined data.
+
+Once the server sends the result to the client it should free up any resources used by the action.
+If the client never asks for the result then the server should discard the result after a timeout period.
+
+### Goal Status Topic
+
+* **Direction**: Server publishes
+* **Content**: List of in-progress goals with: Goal ID, time accepted, and an enum indicating the status
+
+This topic is published by the server to broadcast the status of goals it has accepted.
+The purpose of the topic is for introspection; it is not used by the action client.
+Messages are published when transitions from one status to another occur.
+
+The default QoS settings for a DDS middleware should be TRANSIENT LOCAL with a history size of 1.
+This allows new subscribers to always get the latest state.
+
+The possible statuses are:
+
+* *Accepted*
+  * The goal has been accepted by the action server and may now be executing
+* *Cancelling*
+  * The action server will try to cancel the indicated goal
+* *Cancelled*
+  * The action server successfully canceled the goal
+* *Succeeded*
+  * The action server successfully reached the goal
+* *Aborted*
+  * The action server failed reached the goal
+
+### Feedback Topic
+
+* **Direction**: Server publishes
+* **Content**: Goal id, user defined feedback message
+
+This topic is published by the server to send application specific progress about the goal.
+It is up to the author of the action server to decide how often to publish the feedback.
+
+### Client/Server Interaction Examples
+
+Here are a couple of sequence diagrams depicting typical interactions between an action client and action server.
+
+#### Example 1
+
+In this example, the action client request a goal and gets a response from the server accepting the goal (synchronous).
+Upon accepting the goal, the action server starts a user defined execution method for completing the goal.
+Following the goal request, the client makes an asynchronous request for the result.
+The user defined method publishes feedback to the action client as it executes the goal.
+Ultimately, the user defined method populates a result message that is used as part of the result response.
+
+![Goal Lifecycle Example 0](../img/actions/interaction_example_0.png)
+
+#### Example 2
+
+This example is almost identical to the first, but this time the action  client requests for the goal to be canceled mid-execution.
+Note that the user defined method is allowed to perform any shutdown operations after the cancel request before returning with the cancellation result.
+
+![Goal Lifecycle Example 1](../img/actions/interaction_example_1.png)
+
+#### Example 3
+
+Here is a more complex example involving multiple goals.
+
+TODO
```

Comment:
?

Reply:
I'll add one or two more example sequence diagrams. I wasn't sure if it be too cluttered to be useful; feedback is welcome on any scenarios that might benefit from a sequence diagram for clarity.